dot/parachain/types/subsystems.go

```diff
@@ -0,0 +1,552 @@
+// Copyright 2023 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package parachaintypes
+
+import (
+	"fmt"
+
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+type SubSystemName string
+
+const (
+	CandidateBacking  SubSystemName = "CandidateBacking"
+	CollationProtocol SubSystemName = "CollationProtocol"
+	AvailabilityStore SubSystemName = "AvailabilityStore"
+)
+
+var (
+	_ ProvisionerMessage           = (*PMProvisionableData)(nil)
+	_ ProvisionableData            = (*PDBackedCandidate)(nil)
+	_ ProvisionableData            = (*PDMisbehaviorReport)(nil)
+	_ StatementDistributionMessage = (*SDMBacked)(nil)
+	_ CollatorProtocolMessage      = (*CPMBacked)(nil)
+	_ ProspectiveParachainsMessage = (*PPMCandidateBacked)(nil)
+	_ ProspectiveParachainsMessage = (*PPMIntroduceCandidate)(nil)
+	_ ProspectiveParachainsMessage = (*PPMCandidateSeconded)(nil)
+	_ RuntimeApiMessage            = (*RAMRequest)(nil)
+	_ RuntimeApiRequest            = (*RARValidationCodeByHash)(nil)
+	_ CandidateValidationMessage   = (*CVMValidateFromExhaustive)(nil)
+	_ AvailabilityStoreMessage     = (*ASMStoreAvailableData)(nil)
```

Comment:
I would say let the names be ridiculously long as long as we can understand what they get used for. In this case, it isn't quite clear what PMP, PDB, SDM, CPM, etc are.

We could also use a comment to explain those names, but I prefer that information in the variable name, since that saves a trip to look for a comment explaining the struct.